jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/SecuredRegistrationEndpoint.java

```diff
@@ -54,4 +66,43 @@ public Response registerSecured(@DefaultValue("") @HeaderParam("x-ag-old-token")
 		return register(var, oldToken, entity, synchronously, request);
 	}
 
+	/**
+	 * Endpoint for adding UTR to an existing user at Keycloak.
+	 * Notice: after successful UTR sync user credentials shall be disabled
+	 *
+	 * @param alias User alias
+	 * @param request The request object
+	 *
+	 * return UTR
+	 *
+	 * @statuscode 200 Successful storage of the alias.
+	 * @statuscode 404 if keyclaok user or UTR for the user not found
+	 *
+	 */
+	@POST
+	@Path("/syncUtr/{alias}")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response syncUtr(@PathParam("alias") String alias, @Context HttpServletRequest request) {
+		PushApplication app = authenticationHelper.loadApplicationWhenAuthorized(request);
+		if (!keycloakService.exists(alias, app.getName())) {
+			return Response.status(Response.Status.NOT_FOUND.getStatusCode(), "Keycloak user " + alias + " not found").build();
+		}
+
+		Collection<UserTenantInfo> tenantRelations = aliasService.getTenantRelations(alias);
+		if (tenantRelations.isEmpty()) {
+			return Response.status(Response.Status.NOT_FOUND.getStatusCode(), "UTR for user " + alias + " not found").build();
+		}
+		keycloakService.updateTenantsExistingUser(alias, tenantRelations, app.getName());
+
+		String realmName = keycloakService.getRealmName(app.getName());
+
+		// add 'installation' realm role
+		keycloakService.addUserRealmRoles(Collections.singletonList(keycloakService.KEYCLOAK_ROLE_USER), alias, realmName);
+
+		// disable password
+		keycloakService.disableUserCredentials(alias, realmName);
```

Comment:
consider changing the URL for this method as it does more then just syncing the UTR. 
perhaps rephrase to `bindWithSSO` or something which doesn't give the reader a false sense of what is actually going on